### PR TITLE
Fix Demo Files

### PR DESCRIPTION
**Pull request by mitchute**
Fix issues with demo files

### CODE REVIEW

File: thermalnetwork/network.py [hunk 1]
```diff
@@ -670,15 +674,17 @@ def update_sys_params(self, system_parameter_path: Path, output_directory_path:
         # Load the existing system parameters
         sys_params: dict = json.loads(system_parameter_path.read_text())
         ghe_params: list = sys_params["district_system"]["fifth_generation"]["ghe_parameters"]["ghe_specific_params"]
-        for ghe_id in output_directory_path.iterdir():
-            # each ghe_id is a directory
-            if not ghe_id.is_dir():
-                continue
```
Comment (vtnate) [inline thermalnetwork/network.py:676]: Can we keep this check in? There are sometimes other files that we want to skip over during this loop

File: thermalnetwork/network.py [hunk 2]
```diff
@@ -670,15 +674,17 @@ def update_sys_params(self, system_parameter_path: Path, output_directory_path:
...
-            summary_data_path = ghe_id / "SimulationSummary.json"
+        for ghe in ghe_params:
+            ghe_id = ghe["ghe_id"]
+            summary_data_path = output_directory_path / ghe_id / "SimulationSummary.json"
+
+            if not summary_data_path.exists():
+                logger.error(f"Error sizing GHE: {ghe_id}")
+
```
Comment (mitchute) [inline thermalnetwork/network.py:683]: @vtnate I saw a comment in my inbox about this, but don't see anything here. Anyway, feel free to modify this if needed, but something was definitely wrong with it.

Comment (vtnate) [reply thermalnetwork/network.py:683]: I re-thought my comment and deleted it 😆 .

File: demos/sdk_output_skeleton_13_buildings/run/baseline_scenario/ghe_dir/sys_params_proportional.json [hunk 3]
```diff
@@ -504,7 +504,7 @@
         "pump_flow_rate": 0.01
       },
       "ghe_parameters": {
-        "version": "0.2.5",
+        "version": "0.3.0",
```
Comment (vtnate) [inline demos/sdk_output_skeleton_13_buildings/run/baseline_scenario/ghe_dir/sys_params_proportional.json:507]: Keeping this up to date with our release version suppresses a warning.

File: thermalnetwork/network.py [hunk 4]
```diff
@@ -670,15 +674,17 @@ def update_sys_params(self, system_parameter_path: Path, output_directory_path:
...
         ghe_params: list = sys_params["district_system"]["fifth_generation"]["ghe_parameters"]["ghe_specific_params"]
-        for ghe_id in output_directory_path.iterdir():
-            # each ghe_id is a directory
-            if not ghe_id.is_dir():
-                continue
-            summary_data_path = ghe_id / "SimulationSummary.json"
+        for ghe in ghe_params:
+            ghe_id = ghe["ghe_id"]
+            summary_data_path = output_directory_path / ghe_id / "SimulationSummary.json"
+
```
Comment (vtnate) [inline thermalnetwork/network.py:680]: Nice improvement!

File: thermalnetwork/network.py [hunk 5]
```diff
@@ -732,6 +738,12 @@ def run_sizer_from_cli_worker(
         if feature["properties"]["type"] == "Building" and feature["properties"]["id"] in building_id_list
     ]
 
+    if len(building_features) == 0:
+        logger.error(
+            'No buildings found. Ensure the GeoJSON "Feature" "id" keys match the system \n'
+            'parameter file "geojson_id" key values for the respective buildings.'
+        )
```
Comment (vtnate) [inline thermalnetwork/network.py:745]: Helpful error messages? 🙀

File: thermalnetwork/tests/test_network.py [hunk 6]
```diff
@@ -56,9 +56,7 @@ def test_network_two_ghe_area_proportional(self):
         )
 
         # -- Check
-        expected_depth_of_boreholes = 60
-        # FIXME: 60 is the minimum borehole length for a GHE (as set by sys-param file), which implies the loads
-        # were not correctly sent to GHED.
+        expected_depth_of_boreholes = pytest.approx(130, 1)
```
Comment (vtnate) [inline thermalnetwork/tests/test_network.py:59]: We're getting actual values returned from GHED, so we must be giving it better information now. So we can test a real depth.

**Review by vtnate — APPROVED** [review]
Less worse & more better!

File: demos/sdk_output_skeleton_2_ghe_sequential/run/baseline_scenario/ghe_dir/sys_params.json [hunk 7]
```diff
@@ -38,7 +38,44 @@
...
+        "cop_heat_pump_cooling": 3.5,
+        "ets_pump_flow_rate": 0.0005,
+        "ets_pump_head": 10000,
+        "fan_design_flow_rate": 0.25,
+        "fan_design_head": 150
+      },
+      "photovoltaic_panels": [],
+      "diesel_generators": [],
+      "battery_banks": []
+    },
```
Comment (mitchute) [inline demos/sdk_output_skeleton_2_ghe_sequential/run/baseline_scenario/ghe_dir/sys_params.json:76]: This all was just a copy/paste from above to get this running.

**Comment by mitchute** [comment]
@vtnate the staggered example is throwing the error below.

```
Traceback (most recent call last):
  File "/home/mitchute/Projects/ThermalNetwork/thermalnetwork/network.py", line 894, in <module>
    exit(run_sizer_from_cli())
         ^^^^^^^^^^^^^^^^^^^^
  File "/home/mitchute/Projects/ThermalNetwork/venv/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mitchute/Projects/ThermalNetwork/venv/lib/python3.12/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/mitchute/Projects/ThermalNetwork/venv/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mitchute/Projects/ThermalNetwork/venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mitchute/Projects/ThermalNetwork/thermalnetwork/network.py", line 890, in run_sizer_from_cli
    return run_sizer_from_cli_worker(system_parameter_file, scenario_directory, geojson_file, output_directory)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mitchute/Projects/ThermalNetwork/thermalnetwork/network.py", line 778, in run_sizer_from_cli_worker
    connected_features = network.get_connected_features()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mitchute/Projects/ThermalNetwork/thermalnetwork/network.py", line 88, in get_connected_features
    if feature["properties"]["id"] == [connected_features[0]] and feature["properties"]["type"] != "Building":
                                       ~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```

cli commands

```
-y
demos/sdk_output_skeleton_2_ghe_staggered/run/baseline_scenario/ghe_dir/sys_params.json
-s
demos/sdk_output_skeleton_2_ghe_staggered/run/baseline_scenario
-f
demos/sdk_output_skeleton_2_ghe_staggered/network.geojson
-o
demos/sdk_output_skeleton_2_ghe_staggered/run/baseline_scenario
```

**Comment by vtnate** [comment]
Ugh, ok. That linked issue will fix it.

File: demos/sdk_output_skeleton_2_ghe_staggered/baseline_scenario.csv
Comment (vtnate) [inline demos/sdk_output_skeleton_2_ghe_staggered/baseline_scenario.csv:1]: I created & included this file to make it easy to generate new sys-param files in the future with the GMT

**Comment by mitchute** [comment]
@vtnate I'm done here. Feel free merge it after you review and make any other needed changes.

**Pull request merged**